components/html-block/html-block.js

```diff
@@ -230,10 +240,21 @@ class HtmlBlock extends RtlMixin(LitElement) {
 		`;
 	}
 
-	updated(changedProperties) {
+	async updated(changedProperties) {
 		super.updated(changedProperties);
+		if (changedProperties.has('html') && this.html !== undefined && this.html !== null && !this._hasSlottedContent) {
+			const renderContainer = this.shadowRoot.querySelector('.d2l-html-block-rendered');
+			renderContainer.innerHTML = this.html;
+			await this._processRenderers(renderContainer);
+		}
 		if (this._contextChanged()) {
-			this._render();
+			if (this._hasSlottedContent) this._render();
+			else if (this.html !== undefined && this.html !== null) {
+				const renderContainer = this.shadowRoot.querySelector('.d2l-html-block-rendered');
+				renderContainer.innerHTML = this.html;
+				await this._processRenderers(renderContainer);
```

Comment:
The idea here is to only render the actual results of the `html` property in update. If we try to do it in `render`, MathJax modifies the DOM and Lit can't track whether `html` has actually changed. Doing it this way means we keep Lit's change detection while also preventing the world from exploding post-render.